Add job page to scenario mgt template

jrobinAV opened the pull request:
NEw page on the scenario mgt template to visualize and manage jobs.

jrobinAV committed:
Add job page to scenario mgt template

jrobinAV committed:
increase time out to 40 mins

AlexandreSajus commented:
I just tested this. Looks good, very easy to understand. It would be helpful if the input node had a small default dataset to run the scenario. I just added a 3-line csv dataset and it was impressive to be able to run the scenario from the app and see job "Completed" and the result

AlexandreSajus reviewed (changes requested):
I just tested this. Looks good, very easy to understand. It would be helpful if the input node had a small default dataset to run the scenario. I just added a 3-line csv dataset, and it was impressive to run the scenario from the app and see the job "Completed" and the result.

FlorianJacta commented on src/taipy/templates/scenario-management/{{cookiecutter.__root_folder_name}}/pages/root.md:
```diff
@@ -9,8 +9,13 @@
 |>
 
-<|part|class_name=main|render={selected_scenario}|
+<|part|class_name=main|
```
What is the use of class_name=main

jrobinAV replied:
IDK. Is'nt it used for applying default styling?

FlorianJacta replied:
I don't think so

FlorianJacta commented:
Excellent PR! Few minor points on this template for a future PR:
- we could put margin to 0 in the run to make it nicer looking:
`gui.run(title="{{cookiecutter.__application_title}}", margin="0em")`

![image](https://github.com/Avaiga/taipy-templates/assets/98709993/d8da711d-598a-4066-a5bf-40cea7cb2243)

- we can delete the lines concerning the issue on `from ... import *`

- we could also use `on_submission_change` callback

jrobinAV committed:
Merge branch 'develop' into add_job_page

jrobinAV commented:
> I just tested this. Looks good, very easy to understand. It would be helpful if the input node had a small default dataset to run the scenario. I just added a 3-line csv dataset, and it was impressive to run the scenario from the app and see the job "Completed" and the result.

Actually, we removed the dataset recently.

The purpose of a template is not to demonstrate Taipy features nor to teach users how to do an application. The goal is to speed up the development of an application. If we add a dataset, as a user, I would systematically remove the data set right after scaffolding the application. That would be the very first thing to do, so it's slowing down the development.
Instead of adding a dataset, we have an upload control so the user can directly upload its dataset. We also have placeholders that show where to initialize data.

I don't want to change that.

jrobinAV commented:
> Excellent PR! Few minor points on this template for a future PR:
> 
> * we could put margin to 0 in the run to make it nicer looking:
>   `gui.run(title="{{cookiecutter.__application_title}}", margin="0em")`
> 
> ![image](https://user-images.githubusercontent.com/98709993/273817777-d8da711d-598a-4066-a5bf-40cea7cb2243.png)
> 
> * we can delete the lines concerning the issue on `from ... import *`
> * we could also use `on_submission_change` callback

@FlorianJacta Can you create an improvement ticket for that?

AlexandreSajus commented:
> > I just tested this. Looks good, very easy to understand. It would be helpful if the input node had a small default dataset to run the scenario. I just added a 3-line csv dataset, and it was impressive to run the scenario from the app and see the job "Completed" and the result.
> 
> Actually, we removed the dataset recently.
> 
> The purpose of a template is not to demonstrate Taipy features nor to teach users how to do an application. The goal is to speed up the development of an application. If we add a dataset, as a user, I would systematically remove the data set right after scaffolding the application. That would be the very first thing to do, so it's slowing down the development. Instead of adding a dataset, we have an upload control so the user can directly upload its dataset. We also have placeholders that show where to initialize data.
> 
> I don't want to change that.

Then it needs to be clear that a dataset needs to be uploaded before I run an execution.

I imagine a user will create this template, try running an execution, and fail with an error saying "no dataset found"; the user will be frustrated and not touch the template again.
